zaza/openstack/charm_tests/neutron/tests.py

```diff
@@ -248,6 +249,138 @@ def _get_services(cls):
         return services
 
 
+class NeutronGatewayStatusActionsTest(test_utils.OpenStackBaseTest):
+    """Test status actions of Neutron Gateway Charm.
+
+    actions:
+      * get-status-routers
+      * get-status-dhcp
+      * get-status-lb
+    """
+
+    SKIP_LBAAS_TESTS = True
+
+    @classmethod
+    def setUpClass(cls, application_name='neutron-gateway', model_alias=None):
+        """Run class setup for running Neutron Gateway tests."""
+        super(NeutronGatewayStatusActionsTest, cls).setUpClass(
+            application_name, model_alias)
+        # set up clients
+        cls.neutron_client = (
+            openstack_utils.get_neutron_session_client(cls.keystone_session))
+
+        # Loadbalancer tests not supported on Train and above and on
+        # releases Mitaka and below
+        current_release = openstack_utils.get_os_release()
+        bionic_train = openstack_utils.get_os_release('bionic_train')
+        xenial_mitaka = openstack_utils.get_os_release('xenial_mitaka')
+        cls.SKIP_LBAAS_TESTS = (xenial_mitaka <= current_release or
+                                current_release >= bionic_train)
+
+    def tearDown(self):
+        """Cleanup loadbalancers if there are any left over."""
+        super(NeutronGatewayStatusActionsTest, self).tearDown()
+        if not self.SKIP_LBAAS_TESTS:
+            load_balancers = self.neutron_client.list_loadbalancers().get(
+                'loadbalancers', [])
+            for lbaas in load_balancers:
+                self.neutron_client.delete_loadbalancer(lbaas['id'])
```

Comment:
This gets run after each test in the class?  Is this what you want?  I only see one test that uses load balancers and it might be more "obvious" and less resource hungry to just wrap the majority of a test in a try: except: and put this code in a finally: to ensure it gets run?

Reply:
This is a good point. Indeed the original purpose of this `tearDown` was to ensure that the loadbalancers are removed even if the test that created them failed. I'll fix it.